source/Examples/Resources/nxScript/CreateFileNxScript.ps1

```diff
@@ -0,0 +1,63 @@
+configuration CreateFileNxScript
+{
+    [CmdletBinding()]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [ValidateNotNullOrEmpty()]
+        [String]
+        $FilePath,
+
+        [Parameter(Mandatory = $true)]
+        [ValidateNotNullOrEmpty()]
+        [String]
+        $FileContent
+    )
+
+    Import-DscResource -ModuleName 'nxtools'
+
+    nxScript MyScript
+    {
+        GetScript = {
+            $Reason = @{
+                Code = "Script:Script:FileMissing"
+                Phrase = "File does not exist"
+            }
+
+            if (Test-Path -Path $using:FilePath)
+            {
+                $text = $(Get-Content -Path $using:FilePath -Raw).Trim()
+                if ($text -eq $using:FileContent)
+                {
+                    $Reason.Code = "Script:Script:Success"
+                    $Reason.Phrase = "File exists with correct content"
+                }
+                else
+                {
+                    $Reason.Code = "Script:Script:ContentMissing"
+                    $Reason.Phrase = "File exists but has incorrect content"
+                }
+            }
+
+            return @{
+                Reasons = @($Reason)
```

Comment:
It's possible for this to return no Reasons if the resource is compliant? That's going to error out with this resource even though the resource itself is fine? I know that in Policy we shouldn't really be hitting this unless the resource is non-compliant, but when testing with the GC module can you hit this when the resource is compliant? Just don't want customers to get confused that their resource is erroring when there's nothing wrong with it. Should we return a Reason even when Compliant in our example? Or not error when GetScript doesn't return any Reasons?

Reply:
This example specifically will always return a Reason whether compliant or non-compliant. When I run Get-GuestConfigurationPackageComplianceStatus, it returns the Success reason (line 32) when complianceStatus=true.

However, currently, nxScript allows you to not return any Reasons or any hashtable at all. This is just a design decision to allow users to skip creating Reasons if they don't want to. GetScript will not error, it just won't return any Reasons. What do you think about this? Is this a bad idea?

Reply:
After our discussion, GetScript and SetScript are optional parameters now. If GetScript is defined, it MUST return a hashtable with a Reasons key with at least one Reason object. I think this is less confusing.